### PR TITLE
Expose next retry delay

### PR DESCRIPTION
**Pull request by Sushisource**
## What was changed
Expose the new next retry delay field for activity failures

## Why?
Parity

## Checklist

1. Closes https://github.com/temporalio/sdk-typescript/issues/1349

2. How was this tested:
Added test verifying property is sent to Core, where functionality is already tested

3. Any docs updates needed?

### CODE REVIEW

File: packages/common/src/time-base.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { StringValue } from 'ms';
+
+/**
+ * A duration, expressed either as a number of milliseconds, or as a {@link https://www.npmjs.com/package/ms | ms-formatted string}.
+ */
+export type Duration = StringValue | number;
```
Comment (mjameswh) [inline packages/common/src/time-base.ts:6]: I don't mind the refactor, but why?

File: packages/test/src/test-worker-activities.ts [hunk 2]
```diff
@@ -246,3 +254,31 @@ test('Non ApplicationFailure TemporalFailures thrown from Activity are wrapped w
     t.is(result?.failed?.failure?.applicationFailureInfo?.type, 'TemporalFailure');
   });
 });
+
+test('nextRetryDelay in activity failures is propagated to Core', async (t) => {
```
Comment (mjameswh) [inline packages/test/src/test-worker-activities.ts:258]: Please also add an integration test for this.

File: packages/common/src/time-base.ts (hunk 1 above)
Comment (Sushisource) [reply packages/common/src/time-base.ts:6]: Circular import

Comment (mjameswh) [reply packages/common/src/time-base.ts:6]: I see. Can you try instead to move the `WorkflowExecutionAlreadyStartedError` class out of `errors.ts` to `failure.ts`? I think that should break the cycle in a more natural way, given that `WorkflowExecutionAlreadyStartedError` is indeed a `TemporalFailure`, not a generic Error.

**Review by mjameswh — APPROVED** [review]
Looks great, thanks!

**Pull request merged**